service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -625,6 +630,14 @@ public void removeWorkspaceRoleFromUser(
     if (!roleMembers.contains(targetUserEmail)) {
       return;
     }
+    // Additionally, validate that the user is not removing themselves as the sole owner. WSM does
+    // not allow users to abandon resources this way.
+    if (role.equals(WsmIamRole.OWNER)
```

Comment:
Putting this check into the `RemoveUserFromSamStep` would shrink the window where the result of this check becomes obsolete due to concurrent action.